src/syntax/syntax.rkt

```diff
@@ -71,53 +80,81 @@
 
 (define (operator-info operator field) (table-ref operators operator field))
 
+(define (operator-remove! operator)
+  (table-remove! operators operator)
+  (*loaded-ops* (remove operator (*loaded-ops*))))
```

Comment:
Like with `member`, it's better, I think, to replace `remove` with `set-remove`.